src/wallet/wallet2.cpp

```diff
@@ -8002,15 +8004,35 @@ std::vector<wallet2::pending_tx> wallet2::create_transactions_from(const crypton
           detail::digit_split_strategy, tx_dust_policy(::config::DEFAULT_DUST_THRESHOLD), test_tx, test_ptx);
       auto txBlob = t_serializable_object_to_blob(test_ptx.tx);
       needed_fee = calculate_fee(fee_per_kb, txBlob, fee_multiplier);
-      available_for_fee = test_ptx.fee + test_ptx.dests[0].amount + test_ptx.change_dts.amount;
+      available_for_fee = test_ptx.fee + test_ptx.change_dts.amount;
+      for (auto &dt: test_ptx.dests)
+        available_for_fee += dt.amount;
       LOG_PRINT_L2("Made a " << get_size_string(txBlob) << " tx, with " << print_money(available_for_fee) << " available for fee (" <<
         print_money(needed_fee) << " needed)");
 
+      // add last output, missed for fee estimation
+      if (outputs > 1)
+        tx.dsts.push_back(tx_destination_entry(1, address, is_subaddress));
```

Comment:
For fee estimation purposes, I add (above in the code) `1 if outputs == 1 else outputs - 1` destinations, and here I add the last one, but only if the one added wasn't actually the last one, i.e. `outputs > 1`.

Reply:
I just don't get what this does. Can you explain in detail why you think this is needed ?

Reply:
Sorry it took me so long. Well if I recall correctly, there was a problem with fee estimation that resulted in either creating a dust change output or (in case this dust was distributed among outputs) in insufficient fee error, so the idea was for fee estimation to assume `N - 1` output and change, and than replace change with an actual outputs.

Reply:
I still don't get why it's needed at all.

AFAICT, this change means that if you want to sweep your outputs into, say, 4 different new outputs, you are adding a 5th output here. Am I not reading this right ?

Reply:
No, if you want to sweep into 4 outputs, this code adds 3 outputs, than runs fee estimation and after that adds 4th (instead of a change output from the point of view of fee estimation).

Reply:
OK, I understand now, thanks for explaining. I wasn't seeing it in conjunction with the loop a few lines above.
Now, the problem is that the fee is calculated from the wrong number of outputs in the first place, so adding N-1 outs, then that one later seems wrong. Why not add all outputs normally, then adjust the last amount to account for fee ?

Reply:
Well, this was the (obvious) initial intent, but for an hour of debug I kept having insufficient fee error, no matter how I adjusted last output, and it seemed like a problem with initial estimation, so I decided to go with minimal changes (i.e. not touching fee estimation logic) and used this workaround. If it is suboptimal, I'll spend some more time on it in the next few days

Reply:
Well, it seems wrong AFAICT: it'll estimate a fee for N-1 outs instead of N outs, so you should get an even smaller fee. The bug was likely elsewhere ?

Reply:
Actually, it'll estimate a fee for N-1 outputs and a change output (since we run an initial estimation on outputs of denomination 1 and all the money is assumed to go to change), thus giving us N outputs in total. After that, since this is a sweep operation, we replace change with regular output, ending up with N split outputs, as we wanted it.

About the fee estimation, it's not actually a bug, it's just that the change handling is buried too deep in the transaction construction code (function `transfer_selected()`) to be easily disabled.

Reply:
Alright, since you're tested it, I'll take your word for it.